src/Compilers/Core/CommandLine/BuildClient.cs

```diff
@@ -93,13 +93,14 @@ internal int RunCompilation(IEnumerable<string> originalArguments, BuildPaths bu
                         return HandleResponse(buildResponse, parsedArgs, buildPaths);
                     }
                 }
-                catch
+                catch (OperationCanceledException)
                 {
-                    // It's okay, and expected, for the server compilation to fail.  In that case just fall 
-                    // back to normal compilation. 
+                    return CommonCompiler.Succeeded;
```

Comment:
Filed here: https://github.com/dotnet/roslyn/issues/7866

The reason I got rid of the catch is that it was swallowing the NRE from RunServerCompilation.

I understand swallowing all exceptions in the process of trying to run a server compilation, but I think once we have a result we should not expect exceptions.